app/actions/VSPActions.js

```diff
@@ -14,20 +14,29 @@ export const GETVSP_ATTEMPT = "GETVSP_ATTEMPT";
 export const GETVSP_FAILED = "GETVSP_FAILED";
 export const GETVSP_SUCCESS = "GETVSP_SUCCESS";
 
-export const getVSPInfo = (host) => async (dispatch) => {
+export const getVSPInfo = (host) => (dispatch) => new Promise((resolve, reject) => {
   dispatch({ type: GETVSP_ATTEMPT });
-  try {
-    // add https into vsp host, so we can fetch its information.
-    host = "https://" + host;
-    wallet.allowVSPHost(host);
-    const info = await wallet.getVSPInfo(host);
-    dispatch({ type: GETVSP_SUCCESS, info });
-    return info.data;
-  } catch (error) {
-    dispatch({ type: GETVSP_FAILED, error });
-    return { error };
-  }
-};
+  // 5 seconds for timeout.
+  const TIMEOUT_TIME = 5000;
+  // add https into vsp host, so we can fetch its information.
```

Comment:
string template

Reply:
not sure what you mean here. Move this to constants file?

Reply:
ops was wrong line, see line 22 suggestion.